### ISSUE TITLE
Type naming for `btreetables`

### ISSUE THREAD

**Issue by haxscramper** [nim-lang/fusion#31]
tl;dr - rename `btreetables.Table` & related to something like `BTreeTable`. Why? Easy to confuse with `tables.Table` and leads to name clashes.

Main reason for name change: it is really confusing to have two **identically** named types, which have different asymptotics and behaviour (e.g `log n` + sorted __vs__ unsorted & constant). Using the name naming not only leads to user confusion but also creates severe name clashes for all type declarations and constructor procs.

I would suggest renaming type to something like `TreeTable` / `BTreeTable` (consistent with module name, but probably too long to type) or `BTable` (shorter, but may still be confusing - not too much different from regular `Table`)

**Comment by planetis-m** [comment]
I think the general idea is for `btreetables` to be a drop-in replacement. I'm not convinced how `Map` communicates the difference in time complexity. Is it because `std::map` uses the same data-structure as btreetables? I didn't know that, until recently.

**Comment by haxscramper** [comment]
Yes, it appears that the original idea for `btreetables` was to make a drop-in replacement, but I would argue that this is **not** something that is necessary. The only reason why this might be needed is when a person unknowingly choose data structure with unwanted constraints, and wants to switch to the other one. Otherwise, I fail to see how `O(1), unsorted key-value` and `O(log n), sorted` need to be swapped with each other easily. They have *very* different behaviours both in terms of runtime speed and just in general.

But I can see quite easily how someone might need to use both in the same module, and then they will be facing name clashes all over the place, on each `initTable`, each type declaration involving `Table` and so on.
